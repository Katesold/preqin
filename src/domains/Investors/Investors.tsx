import { memo, useEffect, useState } from "react";

import { StyledTableContainer } from "./styles";
import Table from "../../components/Table/Table";

export const Investors = memo(() => {
  const [firmsData, setFirmsData] = useState<{ data: any[] }>({ data: [] });
  const getData = async () => {
    try {
      const response = await fetch(
        "https://api.preqin.com/api/Investor?firmId=2670,2792,3320,3611",
        {
          headers: {
            "Content-Type": "application/json",
            Authorization: `Bearer ${process.env.REACT_APP_TOKEN}`,
          },
          method: "GET",
        }
      );
      if (response.status === 200) {
        const result = await response.json();
        setFirmsData(result);
      }
    } catch (error) {
      window.alert("Error");
    }
  };

  useEffect(() => {
    getData();
  }, []);

  return (
    <StyledTableContainer>
      <h1>Investors List</h1>
      {firmsData.data.length ? <Table firms={firmsData.data} /> : null}
    </StyledTableContainer>
  );
});
