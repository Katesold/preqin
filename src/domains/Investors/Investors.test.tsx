import { render, screen } from "@testing-library/react";

import { Investors } from "./Investors";

window.alert = jest.fn();

test("renders table container and fetches data", async () => {
  const { container } = render(<Investors />);

  expect(container).toMatchSnapshot();
  expect(screen.queryByText("Firm ID")).not.toBeInTheDocument();
});
