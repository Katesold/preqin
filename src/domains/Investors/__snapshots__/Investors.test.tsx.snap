// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders table container and fetches data 1`] = `
<div>
  <div
    class="sc-beySPh jqVKZZ"
  >
    <h1>
      Investors List
    </h1>
  </div>
</div>
`;
